tests/src/Type/data/inspector.php

```diff
@@ -12,82 +12,146 @@ function mixed_function(): mixed {
 // Inspector::assertAll()
 $callable = fn (string $value): bool => $value === 'foo';
 $input = mixed_function();
-assert(Inspector::assertAll($callable, $input));
-assertType("iterable", $input);
+if (Inspector::assertAll($callable, $input)) {
+    assertType("iterable", $input);
+}
+else {
+    assertType('mixed~iterable', $input);
+}
 
 $input = mixed_function();
 $callable = is_string(...);
-assert(Inspector::assertAll($callable, $input));
-assertType('iterable', $input);
+if (Inspector::assertAll($callable, $input)) {
+    assertType('iterable', $input);
+}
+else {
+    assertType('mixed~iterable', $input);
+}
```

Comment:
@mglaman it's kind of works, but I still don't like it.

```php
$input = mixed_function();
$callable = is_string(...);
if (Inspector::assertAll($callable, $input)) {
    assertType('iterable', $input);
} else {
    assertType('mixed~iterable', $input);
}
```

Actually, it doesn't mean that `$input` is not `iterable` after that `::assertAll()`. It just means it does not contain strings, but it can still be iterable... I think in `else`, it should be `mixed`, as we can't guarantee it is not iterable. We can't narrow it down to `mixed~iterable<string>`, so we better fallback to `mixed`.

Reply:
I see what you mean, but it is never intended to be used in conditionals like this – it's meant to be used in `asserts` which fail if `false`. So I think it's OK that the condition fails. Because none of the type specifying `assert` extensions would work